FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h

```diff
@@ -0,0 +1,42 @@
+#import <Foundation/Foundation.h>
+
+#import "FIRLibrary.h"
+
+@class FIRApp;
+@class FIRSegmentation;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Provides and creates instances of Segmentation. Used in the
+/// interop registration process to keep track of Segmentation instances for each `FIRApp` instance.
+@protocol FIRSegmentationProvider
+
+/// Cached instances of Segmentation objects.
+@property(nonatomic, strong) NSMutableDictionary<NSString *, FIRSegmentation *> *instances;
+
+/// Default method for retrieving a Segmentation instance, or creating one if it doesn't exist.
+- (FIRSegmentation *)segmentation;
+
+@end
+
+/// A concrete implementation for FIRSegmentationInterop to create Segmentation instances and
```

Comment:
Is there a need for FIRSegmentationInterop?

Reply:
Is there a reason not to? What is the alternative? I thought all products were moving to using interop.

Reply:
Interop is only needed for products that want to manage their interface to other Firebase dependencies separately from their library implementation.  Will Segmentation have Firebase pod dependencies?

Reply:
Yes, Segmentation will need to depend on IID and FIS.